sphinx/ext/intersphinx/_load.py

```diff
@@ -179,39 +182,43 @@ def fetch_inventory_group(
     now: int,
 ) -> bool:
     cache_time = now - app.config.intersphinx_cache_limit * 86400
+
+    updated = False
     failures = []
-    try:
-        for inv in invs:
-            if not inv:
-                inv = posixpath.join(uri, INVENTORY_FILENAME)
-            # decide whether the inventory must be read: always read local
-            # files; remote ones only if the cache time is expired
-            if '://' not in inv or uri not in cache or cache[uri][1] < cache_time:
-                safe_inv_url = _get_safe_url(inv)
-                inv_descriptor = name or 'main_inventory'
-                LOGGER.info(__("loading intersphinx inventory '%s' from %s..."),
-                            inv_descriptor, safe_inv_url)
-                try:
-                    invdata = fetch_inventory(app, uri, inv)
-                except Exception as err:
-                    failures.append(err.args)
-                    continue
-                if invdata:
-                    cache[uri] = name, now, invdata
-                    return True
-        return False
-    finally:
-        if not failures:
-            pass
-        elif len(failures) < len(invs):
-            LOGGER.info(__('encountered some issues with some of the inventories,'
-                           ' but they had working alternatives:'))
-            for fail in failures:
-                LOGGER.info(*fail)
-        else:
-            issues = '\n'.join(f[0] % f[1:] for f in failures)
-            LOGGER.warning(__('failed to reach any of the inventories '
-                              'with the following issues:') + '\n' + issues)
+
+    for location in invs:
+        inv: str = location or posixpath.join(uri, INVENTORY_FILENAME)
```

Comment:
Should this be:

```suggestion
        inv: str | None = location or posixpath.join(uri, INVENTORY_FILENAME)
```

?

(I think inventory location targets [can be `None`](https://github.com/sphinx-doc/sphinx/blob/bf57d21fcd3b1e2a8ade0b982eb936cb89bb8726/sphinx/ext/intersphinx/_load.py#L109-L110)?)

Reply:
Oops; perhaps this is better:

```suggestion
        inv: InventoryLocation = location or posixpath.join(uri, INVENTORY_FILENAME)
```

Reply:
It can be None, but now it's not None anymore. Being None means to use `posixpath.join(uri, INVENTORY_FILENAME)` (so here, inv is really a string)

Reply:
They can, but note None is falsy & the uri/objects.inv path is used instead, so inv is always str.